### ISSUE TITLE
Interrupted BackupPC_migrateV3toV4 leads to migrated backups without pc/host/nnn/backupInfo file and missing inodeLast info in pc/host/backups

### ISSUE THREAD

**Issue by c0deright** [backuppc/backuppc#529]
## Migration

I did a migration from `v3.3.2` to `v4.4.0` and used `BackupPC_migrateV3toV4` according to the documentation.

After testing with one host I used `BackupPC_migrateV3toV4 -a` to migrate all hosts which was super slow since it did all the conversations one after each other.

I interrupted the `BackupPC_migrateV3toV4 -a` process and started a `BackupPC_migrateV3toV4 -h host...` processes for each host in parallel instead which did speedup the migration considerable.

## Problem

The interrupted `BackupPC_migrateV3toV4 -a` process did migrate some but not all backups of one host before being interrupted. Let's call that host `foobar`.

Let's assume host `foobar` had 4 backups 0, 1, 2, 3 in `v3` format and `BackupPC_migrateV3toV4` migrated backups 0 and 1 successfully before being interrupted.

Running `BackupPC_migrateV3toV4 -h foobar` again would migrate backups 2 and 3 and at the end, just before `BackupPC_migrateV3toV4` exits, it would create files `pc/foobar/2/backupInfo`, `pc/foobar/3/backupInfo` and update `pc/foobar/backups` with inodeLast information in the last row.

Already migrated backup 0 and 1 would be skipped although the files `pc/foobar/0/backupInfo` and `pc/foobar/1/backupInfo` are missing and `pc/foobar/backups` was not updated with inodeLast for the two backups.

## Solution

`BackupPC_migrateV3toV4` should create the `backupInfo` file and update the `backups` file whenever it finishes migrating a single backup instead of doing this only at the end.

Imagine a host with 0..999 backups and `BackupPC_migrateV3toV4` is being interrupted while working on backup 999. Instead of having 998 correctly migrated backups with their metadata intact we have 998 migrated backups without metadata and running `BackupPC_migrateV3toV4` again for this host won't fix this.

## Workaround

As I understand the code the `pc/host/backups` file holds the info of all backups' `backupInfo` files so `pc/host/NUM/backupInfo` could be reconstructed except for the missing `inodeLast` data.

How can this easily be fixed?

**Comment by GWHAYWOOD** [comment]
Closing as I think this is unlikely to be fixed.

The migration script has never been thoroughly exercised and debugged, and now that installations of version 3 of BackupPC are getting rarer I doubt if anyone will want to put effort into doing that.

If I were ever to use BackupPC_migrateV3toV4 (which I think is most unlikely) I would make sure that firstly the V3 pool on which I was working was a copy, not the original, and secondly that the script's operation was not interrupted.
